drivers/serial/uart_npcx.c

```diff
@@ -48,6 +48,7 @@ struct uart_npcx_data {
 	/* Baud rate */
 	uint32_t baud_rate;
 	struct miwu_dev_callback uart_rx_cb;
+	struct k_spinlock lock;
```

Comment:
Could this do atomic accesses to the register rather than using an explicit spinlock? There's not much benefit if we assume the driver will never run on a chip with SMP, but we can still avoid blocking all interrupts for a moment.

Reply:
> Could this do atomic accesses to the register rather than using an explicit spinlock? There's not much benefit if we assume the driver will never run on a chip with SMP, but we can still avoid blocking all interrupts for a moment.

Do you mean using [atomic services apis](https://docs.zephyrproject.org/apidoc/latest/group__atomic__apis.html) for accessing UFTCTL register? If so, the type of target, `atomic_t`, might not be appropriate for an 8-bit register. My conclusion could be wrong. If you have further comments, please let me know.

Reply:
Thinking more in terms of intrinsics like `__atomic_fetch_and`, but that's probably not portable enough to use directly. `stdatomic.h` might be usable, but it's not very important; a spinlock will do fine as well.